### DISCUSSION TITLE
android 4.2.92 版本崩溃

### DISCUSSION THREAD

**Discussion by chqhu160** [Tencent/libpag#2203] [General]
libpag:4.2.92出现崩溃

Revision: '0'
ABI: 'arm64'
Timestamp: 2024-03-17 15:39:30+0800
pid: 18362, tid: 21286, name: pag-renderer  >>> xx.xxx.xxx <<<
uid: 10140
signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x75e9e2d3ebd0
    x0  0000000000000000  x1  0000f5e90eb403b0  x2  0000f5e90eb40410  x3  0000000000000010
    x4  0000000000000010  x5  0000000000001459  x6  0000f5ec59470010  x7  000000000027be68
    x8  b40075e9e2d3ebc0  x9  000000000000074c  x10 0000000000000001  x11 0000000000000000
    x12 0000000000000000  x13 0000000000000001  x14 0000000000000001  x15 0000f5e90eb40228
    x16 0000f5e910036940  x17 0000f5ec54350ad4  x18 0000000000003210  x19 b400f5ea72c2ec30
    x20 0000f5e90eb40410  x21 0000000000000001  x22 0000f5e90eb41000  x23 0000f5e90eb403a0
    x24 b400f5e9e2d3d180  x25 b400f5ea12c26c10  x26 0000f5e90eb41000  x27 0000279c000047ba
    x28 0000000000000001  x29 0000f5e90eb403f0
    lr  0000f5e90fe87b48  sp  0000f5e90eb40390  pc  0000f5e90fe87b24  pst 0000000020001000

backtrace:
      #00 pc 000000000012eb24  /system/app/DemoTest/lib/arm64/libpag.so (BuildId: 6047046b0a383195afd01c8d8a280898a6b8f755)
      #01 pc 000000000012ea80  /system/app/DemoTest/lib/arm64/libpag.so (BuildId: 6047046b0a383195afd01c8d8a280898a6b8f755)
      #02 pc 00000000000a7f1c  /system/app/DemoTest/lib/arm64/libpag.so (BuildId: 6047046b0a383195afd01c8d8a280898a6b8f755)
      #03 pc 000000000009e8f8  /system/app/DemoTest/lib/arm64/libpag.so (pag::PAGSurface::draw(pag::RenderCache*, std::__ndk1::shared_ptr<pag::Graphic>, pag::BackendSemaphore*, bool)+660) (BuildId: 6047046b0a383195afd01c8d8a280898a6b8f755)
      #04 pc 000000000009d284  /system/app/DemoTest/lib/arm64/libpag.so (pag::PAGPlayer::flushInternal(pag::BackendSemaphore*)+192) (BuildId: 6047046b0a383195afd01c8d8a280898a6b8f755)
      #05 pc 000000000009d18c  /system/app/DemoTest/lib/arm64/libpag.so (pag::PAGPlayer::flush()+88) (BuildId: 6047046b0a383195afd01c8d8a280898a6b8f755)
      #06 pc 00000000000fff18  /system/app/DemoTest/lib/arm64/libpag.so (Java_org_libpag_PAGPlayer_flushAndFenceSync+204) (BuildId: 6047046b0a383195afd01c8d8a280898a6b8f755)
      #07 pc 000000000113a538  /system/app/DemoTest/oat/arm64/DemoTest.odex (art_jni_trampoline+152) (BuildId: 81f03538dc8eb5d5c573e53cedf46ecdcfc488df)
      #08 pc 000000000113b0cc  /system/app/DemoTest/oat/arm64/DemoTest.odex (org.libpag.PAGPlayer.flush+44) (BuildId: 81f03538dc8eb5d5c573e53cedf46ecdcfc488df)
      #09 pc 00000000011742c0  /system/app/DemoTest/oat/arm64/DemoTest.odex (org.libpag.PAGView.flush+304) (BuildId: 81f03538dc8eb5d5c573e53cedf46ecdcfc488df)
      #10 pc 000000000113d8c8  /system/app/DemoTest/oat/arm64/DemoTest.odex (org.libpag.PAGView$g.handleMessage+792) (BuildId: 81f03538dc8eb5d5c573e53cedf46ecdcfc488df)
      #11 pc 0000000000639a04  /system/framework/arm64/boot-framework.oat (android.os.Handler.dispatchMessage+180) (BuildId: dd3a8c34a0a9e77c3c8efbd10a519450a81928dd)
      #12 pc 000000000063cefc  /system/framework/arm64/boot-framework.oat (android.os.Looper.loop+1516) (BuildId: dd3a8c34a0a9e77c3c8efbd10a519450a81928dd)
      #13 pc 000000000063bb80  /system/framework/arm64/boot-framework.oat (android.os.HandlerThread.run+544) (BuildId: dd3a8c34a0a9e77c3c8efbd10a519450a81928dd)
      #14 pc 0000000000133564  /apex/com.android.art/lib64/libart.so (art_quick_invoke_stub+548) (BuildId: 08172c860b380a8c871ee5b3f08e4ef6)
      #15 pc 00000000001a8a78  /apex/com.android.art/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+200) (BuildId: 08172c860b380a8c871ee5b3f08e4ef6)
      #16 pc 0000000000557988  /apex/com.android.art/lib64/libart.so (art::JValue art::InvokeVirtualOrInterfaceWithJValues<art::ArtMethod*>(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, art::ArtMethod*, jvalue const*)+460) (BuildId: 08172c860b380a8c871ee5b3f08e4ef6)
      #17 pc 00000000005a6d24  /apex/com.android.art/lib64/libart.so (art::Thread::CreateCallback(void*)+1308) (BuildId: 08172c860b380a8c871ee5b3f08e4ef6)
      #18 pc 00000000000afd4c  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+64) (BuildId: f60e9508c5bfe39f73dfa8a2be262d62)
      #19 pc 0000000000050288  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64) (BuildId: f60e9508c5bfe39f73dfa8a2be262d62)

**Comment by CodeJhF** [comment]
抱歉这条当时没有被跟进到，是我们漏了。

因长期无回复，且 libpag 4.3 及更早版本已停止维护（当前仅维护 4.4 / 4.5），先关闭本条。
上报版本：4.2.92。

如果问题在最新版本（4.4 / 4.5）仍能复现，欢迎在本帖回复或新开 issue，并附上 libpag 版本号、平台与系统版本、可复现的 PAG 文件或最小 demo，我们会优先跟进。（stale-close）
